docs/reference/grpc-sync-service.md

```diff
@@ -0,0 +1,41 @@
+---
+description: flagd as a gRPC sync service
+---
+
+# Overview
+
+flagd can expose a gRPC sync service, allowing in-process providers to obtain their flag configurations.
+This mode is **disabled** by default, and you can enable it by using startup flag `--sync-enabled` (or `-e` shorthand flag).
+The gRPC sync stream contains flag configurations currently configured at flagd as [sync-configurations](./sync-configuration.md).
+
+```mermaid
+---
+title: gRPC sync
+---
+erDiagram
+    flagd ||--o{ "sync (file)" : watches
+    flagd ||--o{ "sync (http)" : polls
+    flagd ||--o{ "sync (grpc)" : "sync.proto (gRPC/stream)"
+    flagd ||--o{ "sync (kubernetes)" : watches
+    "In-Process provider" ||--|| flagd : "gRPC sync stream (default port 8015)"
+```
+
+You may change the default port of the service using startup flag `--sync-port` (or `-g` shothand flag).
+
+By default, the gRPC stream expose all the flag configurations merged following merge strategy of flagd.
+You can read more about the merge strategy in our dedicated [concepts guide on syncs](../concepts/syncs.md).
+
+If you specify a `selector` in the gRPC sync request, the gRPC service will attempt match the provided selector valur to a source and stream specific flags.
+For example, if `selector` is set to `myFlags.json`, service will stream flags observed from `myFlags.json` file.
```

Comment:
Does the selector have to match the CLI arg exactly? If so, we should probably make it `file://myFlags.json` for clarity.

Reply:
Good question, I had to double check on this. 

No, this doesn't have to match CLI arg of URI format. It's only the source. For example if `--uri=file://myFlags.json` is used, then selector should be `myFlags.json`.